### PR TITLE
feat(core): check application custom domain conflict

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Check if a domain is already in use when creating custom domain for protected app.

## Testing

Unit tests.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/queries/application.ts [hunk 1]
```diff
@@ -138,6 +138,19 @@ export const createApplicationQueries = (pool: CommonQueryMethods) => {
...
+   * Find an protected application by its custom domain.
+   * the domain is stored in the `customDomains` field of the `protectedAppMetadata` field.
+   */
+  const findApplicationByProtectedAppCustomDomain = async (domain: string) =>
+    pool.maybeOne<Application>(sql`
+      select ${sql.join(Object.values(fields), sql`, `)}
+      from ${table}
+      where ${fields.protectedAppMetadata} ? 'customDomains'
+      and ${fields.protectedAppMetadata}->'customDomains' @> ${sql.jsonb([domain])}
+      and ${fields.type} = ${ApplicationType.Protected}
```
Comment (simeng-li) [inline packages/core/src/queries/application.ts:151]: do we have index for this jsonb field

File: packages/core/src/routes/applications/application-protected-app-metadata.ts [hunk 2]
```diff
@@ -51,7 +56,13 @@ export default function applicationProtectedAppMetadataRoutes<T extends AuthedRo
         'domain.limit_to_one_domain'
       );
 
-      // TODO: LOG-8066 check if domain is already in use
+      assertThat(
+        !(await findApplicationByProtectedAppCustomDomain(domain)),
+        new RequestError({
+          code: 'domain.hostname_already_exists',
+          status: 409,
```
Comment (simeng-li) [inline packages/core/src/routes/applications/application-protected-app-metadata.ts:63]: need to double-check,  we might use 422 on other resource exists exceptions.

File: packages/core/src/queries/application.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/queries/application.ts:151]: The field `customDomains` is optional, not sure if we can add index directly, maybe add the index later? Created LOG-8072 on this project to track.

File: packages/core/src/routes/applications/application-protected-app-metadata.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/applications/application-protected-app-metadata.ts]: Updated with 422 to align with other exceptions.

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**